tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -641,44 +641,46 @@ def sync_volumes(volumes, index, vol_options, sync_ttl):
             )
 
             etcd_utils.write(vol_brick_path, "")
-
             brick = NS.tendrl.objects.GlusterBrick(
                 NS.tendrl_context.integration_id,
                 NS.node_context.fqdn,
-                brick_name.split(":_")[-1],
-                name=brick_name,
-                vol_id=volumes['volume%s.id' % index],
-                sequence_number=b_index,
-                brick_path=volumes[
-                    'volume%s.brick%s.path' % (index, b_index)
-                ],
-                hostname=volumes.get(
-                    'volume%s.brick%s.hostname' % (index, b_index)
-                ),
-                port=volumes.get(
-                    'volume%s.brick%s.port' % (index, b_index)
-                ),
-                vol_name=volumes['volume%s.name' % index],
-                used=True,
-                node_id=NS.node_context.node_id,
-                status=volumes.get(
-                    'volume%s.brick%s.status' % (index, b_index)
-                ),
-                filesystem_type=volumes.get(
-                    'volume%s.brick%s.filesystem_type' % (index, b_index)
-                ),
-                mount_opts=volumes.get(
-                    'volume%s.brick%s.mount_options' % (index, b_index)
-                ),
-                utilization=brick_utilization.brick_utilization(
-                    volumes['volume%s.brick%s.path' % (index, b_index)]
-                ),
-                client_count=volumes.get(
-                    'volume%s.brick%s.client_count' % (index, b_index)
-                ),
-                is_arbiter=volumes.get(
-                    'volume%s.brick%s.is_arbiter' % (index, b_index)
-                ),
+                brick_dir=brick_name.split(":_")[-1]
```

Comment:
So the only change is you are assigning attributes outside of the constructor?

Reply:
no i am loading brick before saving

Reply:
when see do save directly other brick device details are beomces None. So it happens in each sync, brick details are pushed and again modified to None.

Reply:
Handle etcd.EtcdKeyNotFound since you are loading

Reply:
ok